File: C_Programming/Practise_Programs/PP_032.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_037
//
//  Description       : This program accepts a frequency from the
//                      user and displays numbers from the given
//                      frequency down to 0.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 08/07/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

//////////////////////////////////////////////////////////////////
//
//  Function Name     : Display()
//
//  Description       : It is used to display numbers from the
//                      given number down to 0. It also checks
//                      whether the given input is negative.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 08/07/2026
//
//////////////////////////////////////////////////////////////////

void Display(int iNo)
{
    int iCnt = 0;

    if(iNo < 0)
    {
        printf("Invalid input");
        return;
    }

    for(iCnt = iNo; iCnt >= 0; iCnt--)
    {
        printf("%d", iCnt);
    }
}

int main()
{
    int iValue = 0;

    printf("Enter the frequency : ");
    scanf("%d", &iValue);

    Display(iValue);

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      Enter the frequency : 5
//
//  Sample Output :
//      543210
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_032.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_032
//
//  Description       : This program accepts a number from the user
//                      and counts the number of even and odd digits
//                      present in the given number.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 10/07/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class DigitX
{
    //////////////////////////////////////////////////////////////////
    //
    //  Method Name      : CountEvenOddDigits()
    //
    //  Description      : It is used to count the number of even
    //                     and odd digits present in the given
    //                     number by extracting each digit and
    //                     checking whether it is even or odd.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 10/07/2026
    //
    //////////////////////////////////////////////////////////////////

    public void CountEvenOddDigits(int iNo)
    {
        int iDigit = 0;
        int iCountEven = 0, iCountOdd = 0;

        while(iNo != 0)
        {
            iDigit = iNo % 10;

            if(iDigit % 2 == 0)
            {
                iCountEven++;
            }
            else
            {
                iCountOdd++;
            }

            iNo = iNo / 10;
        }

        System.out.println("Number of even digits : " + iCountEven);
        System.out.println("Number of odd digits : " + iCountOdd);
    }
}

class PP_032
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        DigitX dobj = new DigitX();

        int iValue = 0;

        System.out.println("Enter Number : ");
        iValue = sobj.nextInt();

        dobj.CountEvenOddDigits(iValue);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      Enter Number :
//      123456
//
//  Sample Output :
//      Number of even digits : 3
//      Number of odd digits : 3
//
//////////////////////////////////////////////////////////////////
